src/helpers/helpers.dom.js

```diff
@@ -59,8 +59,7 @@ function getPositionedStyle(styles, style, suffix) {
 
 const useOffsetPos = (x, y, target) => (x > 0 || y > 0) && (!target || !target.shadowRoot);
 
-function getCanvasPosition(evt, canvas) {
-  const e = evt.native || evt;
```

Comment:
What's the reason for no longer preferring the native event?

Reply:
My understanding the code that's currently in master:

* `getCanvasPosition` prefers the native event, like you said - in other words, it ensures that, regardless of whether it's given a `ChartEvent` or a native `Event`, it starts its operations from the native x, y coordinates.
* `getCanvasPosition` is only used by helpers.dom.js's version of `getRelativePosition`. To summarize  and maybe oversimplify, it starts from the native x, y coordinates and calculates the x, y coordinates relative to the chart.
* core.interaction.js's version of `getRelativePosition` to check if it's given a `ChartEvent` and, if so, to return the x, y coordinates from the `ChartEvent` - in other words, if it's given a `ChartEvent` instead of a native `Event`, it assumes / treats the `ChartEvent`'s x, y as already giving the x, y coordinates relative to the chart.

So helpers.dom.js's `getRelativePosition` supports being given both a `ChartEvent` and a native `Event`, and if it's given a `ChartEvent`, it ignores the `ChartEvent`'s relative x, y coordinates and (re)calculates the relative coordinates starting from the native `Event`. To consolidate code and make it easier to export, I'm changing helpers.dom.js to have the same "use `ChartEvent`'s x, y" logic as core.interaction.js's version; after that change, `getCanvasPosition` is only ever given a native `Event`, so it no longer needs to check if it's given a `ChartEvent`.

I added some JSDoc types to help make this clearer. To keep the lint-tsc check happy, I had to add a `@ts-ignore`, because the current `getCanvasPosition` code relies on some loose conversions between `Event`, `TouchEvent`, and `Touch` that would require several changes to make TypeScript happy. I'm not sure if that fits with Chart.js's coding standards; if you want me to back that out or clean up the type handling to remove `@ts-ignore`, please let me know.

If I've misunderstood any of the above - i.e., if `ChartEvent'`s x, y coordinates are ever not relative to the chart - then this won't work. The helpers.dom.tests.js test is to help ensure that this is valid.

Reply:
You're understanding that the `ChartEvent` has the relative position is correct and re-using that data makes sense.

I think the `ts-ignore` is fine for now. Some of these helper functions could be converted directly to Typescript in the future and we could remove the ignores then